### PR TITLE
ML-023/add movie controller

### PR DESCRIPTION
**Pull request by WhatWasThatt**

### CODE REVIEW

File: db/migrate/20220220114730_create_join_table_countries_movies.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class CreateJoinTableCountriesMovies < ActiveRecord::Migration[6.1]
+  def change
+    create_join_table :countries, :movies do |t|
+      # t.index [:country_id, :movie_id]
```
Comment (ShrewHools) [inline db/migrate/20220220114730_create_join_table_countries_movies.rb:4]: comments should be removed)
you really don't need an index?)

File: app/models/movie.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class Movie < ApplicationRecord
+  has_and_belongs_to_many :companies
+  has_and_belongs_to_many :countries
+  has_and_belongs_to_many :genres
```
[the remark this answers is not in the document]
Comment (WhatWasThatt) [reply app/models/movie.rb:4]: has_and_belongs_to_many :genres, -> { distinct }?
or 
t.index [:genre_id, :movie_id], :unique => true?

[the remark this answers is not in the document]
Comment (ShrewHools) [reply app/models/movie.rb:4]: -> distinct = does not allow keeping the database clean) (in this case)

or 
movie.rb 
```
has_many :genres_movies, dependent: :destroy
has_many :genres, through: :genres_movies
```
genres_movie.rb
```
belongs_to :movie
belongs_to :genre
validates :movie_id, uniqueness: { scope: :genre_id, message: '
anything useful' }
```

[the remark this answers is not in the document]
Comment (WhatWasThatt) [reply app/models/movie.rb:4]: Maybe make sense to validate on the database level? Can the genres_movies model be redundant? What do u think about it?

**Review by ShrewHools — APPROVED** [review]
gj

**Review by twill11 — APPROVED** [review]
good work!

**Pull request merged**